demos/http/http_demo_mutual_tls/http_demo_mutual_tls.c

```diff
@@ -0,0 +1,957 @@
+     * during polling. If either holds true, read the available data. */
+    if( ( bytesAvailableToRead > 0 ) || ( pollStatus > 0 ) )
+    {
+        bytesReceived = ( int32_t ) SSL_read( pNetworkContext->pSslContext, pBuffer, bytesToRecv );
+    }
+    /* Poll timed out. */
+    else if( pollStatus == 0 )
+    {
+        LogInfo( ( "Poll timed out and there is no data to read from the buffer." ) );
+    }
+    else
+    {
+        LogError( ( "Poll returned with status = %d.", pollStatus ) );
+        bytesReceived = -1;
+    }
+
+    return bytesReceived;
+}
+
+/*-----------------------------------------------------------*/
+
+static int sendHttpRequest( const HTTPTransportInterface_t * pTransportInterface,
+                            const char * pMethod,
+                            size_t methodLen,
+                            const char * pPath,
+                            size_t pathLen )
+{
+    /* Return value of this method. */
+    int returnStatus = EXIT_SUCCESS;
+
+    /* Configurations of the initial request headers that are passed to
+     * #HTTPClient_InitializeRequestHeaders. */
+    HTTPRequestInfo_t requestInfo;
+    /* Represents a response returned from an HTTP server. */
+    HTTPResponse_t response;
+    /* Represents header data that will be sent in an HTTP request. */
+    HTTPRequestHeaders_t requestHeaders;
+
+    /* Return value of all methods from the HTTP Client library API. */
+    HTTPStatus_t httpStatus = HTTP_SUCCESS;
+
+    assert( pMethod != NULL );
+    assert( methodLen > 0 );
+
+    /* Initialize all HTTP Client library API structs to 0. */
+    ( void ) memset( &requestInfo, 0, sizeof( requestInfo ) );
+    ( void ) memset( &response, 0, sizeof( response ) );
+    ( void ) memset( &requestHeaders, 0, sizeof( requestHeaders ) );
+
+    /* Initialize the request object. */
+    requestInfo.pHost = IOT_CORE_ENDPOINT;
+    requestInfo.hostLen = IOT_CORE_ENDPOINT_LENGTH;
+    requestInfo.method = pMethod;
+    requestInfo.methodLen = methodLen;
+    requestInfo.pPath = pPath;
+    requestInfo.pathLen = pathLen;
+
+    /* Set "Connection" HTTP header to "keep-alive" so that multiple requests
+     * can be sent over the same established TCP connection. */
+    requestInfo.flags = HTTP_REQUEST_KEEP_ALIVE_FLAG;
+
+    /* Set the buffer used for storing request headers. */
+    requestHeaders.pBuffer = userBuffer;
+    requestHeaders.bufferLen = USER_BUFFER_LENGTH;
+
+    httpStatus = HTTPClient_InitializeRequestHeaders( &requestHeaders,
+                                                      &requestInfo );
+
+    if( httpStatus == HTTP_SUCCESS )
+    {
+        /* Initialize the response object. The same buffer used for storing
+         * request headers is reused here. */
+        response.pBuffer = userBuffer;
+        response.bufferLen = USER_BUFFER_LENGTH;
+
+        LogInfo( ( "Sending HTTP %.*s request to %.*s%.*s...",
```

Comment:
IotLogDebug to match the other demos?

Reply:
We have this as LogInfo for all the other demos. The LogDebug only applies when printing the request headers and body.

Reply:
oh right I looked at this wrong thanks